src/helm/benchmark/adaptation/adapter_spec.py

```diff
@@ -68,6 +68,9 @@ class AdapterSpec:
     # set of training instances.  Used to compute error bars.
     num_train_trials: int = 1
 
+    # Randomly sample train examples or use them in order
```

Comment:
To make this clearer: "If true, randomly sample N training examples; if false, select the first N training examples"

Reply:
Edit: it seems like it's not the first N examples, there's still variation defined by seed.  Can you provide a description of why we need this non-traditional sampling strategy?

Reply:
The seed is set to be the `train_trial_index` so it is sequential. The seed was a default parameter to `sample_examples`. When `sample_train` is False, we just set it to be the `train_trial_index` and then use that to index into the train examples. See [here](https://github.com/stanford-crfm/helm/blob/e91e50e77c8237be489196d129a9bb973f3ca8cf/src/helm/benchmark/adaptation/adapters/in_context_learning_adapter.py#L133).

Reply:
I guess the question is not how `sample_train = False` works, but why do you need it, as opposed to just going with the default sampling strategy?